docker/pulumi/Dockerfile

```diff
@@ -131,13 +118,28 @@ RUN ln -s /usr/local/share/pypoetry/bin/poetry /usr/local/bin/
 # poetry will create virtual environments using the python version used by poetry itself.
 RUN poetry config virtualenvs.prefer-active-python true
 
+# Install default nodejs versions and associated tools
+RUN curl -fsSL https://fnm.vercel.app/install | bash -s -- --install-dir "/usr/local/share/fnm" --skip-shell && \
+  ln -s /usr/local/share/fnm/fnm /usr/local/bin/fnm
+ENV FNM_COREPACK_ENABLED="true"
+ENV FNM_VERSION_FILE_STRATEGY="recursive"
+ENV FNM_DIR=/usr/local/share/fnm
+RUN fnm install 18 && \
+  fnm install 20 && \
+  fnm install 22 && \
+  fnm alias 18 default
+ENV PATH=/usr/local/share/fnm/aliases/default/bin:$PATH
+RUN corepack install --global pnpm yarn
+
 # Passing --build-arg PULUMI_VERSION=vX.Y.Z will use that version
 # of the SDK. Otherwise, we use whatever get.pulumi.com thinks is
 # the latest
 ARG PULUMI_VERSION
 
 # Install the Pulumi SDK, including the CLI and language runtimes.
-RUN curl -fsSL https://get.pulumi.com/ | bash -s -- --version $PULUMI_VERSION && \
+# TODO: Should use $PULUMI_VERSION, not dev
+# RUN curl -fsSL https://get.pulumi.com/ | bash -s -- --version $PULUMI_VERSION && \
+RUN curl -fsSL https://get.pulumi.com | sh -s -- --version dev && \
```

Comment:
TODO: before merging this PR switch back to the proper version release after the next CLI release